### PR TITLE
update Electron to 1.8.3

### PR DESCRIPTION
**Pull request by iAmWillShepherd**
Updates the app to use electron 1.8.2 and updates related dependencies to latest versions.

- [x] Test build
  - [x] macOS
  - [x] Windows
  - [x] Linux

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -128,10 +128,10 @@
     "@types/uuid": "^3.4.0",
     "@types/winston": "^2.2.0",
     "@types/xml2js": "^0.4.0",
-    "electron": "1.7.11",
-    "electron-builder": "19.48.3",
+    "electron": "1.8.2",
```
Comment (shiftkey) [inline package.json]: You need to bump `app/.npmrc` to this same version to ensure native node modules are built correctly.

File: package.json [hunk 2]
```diff
@@ -128,10 +128,10 @@
...
+    "electron-builder": "19.56.0",
```
Comment (shiftkey) [inline package.json]: Be mindful that we only use this for the Linux installer creation (which isn't done by CI currently).

File: package.json (hunk 1 above)
Comment (iAmWillShepherd) [reply package.json]: Yes, I spent about 30 mins figuring that out! 👍

File: package.json (hunk 2 above)
Comment (iAmWillShepherd) [reply package.json]: ok, I will revert this back to the previous version to see if it plays nice with Electron 1.8.

**Comment by shiftkey** [comment]
Don't forget to nuke the `package-lock.json` file :boom::boom::boom:

**Comment by shiftkey** [comment]
I'd also recommend cutting a test release and validating the installer works for these platforms

**Comment by iAmWillShepherd** [comment]
I released a test release in central (everything seems to work 🤞).

**Comment by iAmWillShepherd** [comment]
🔢

**Comment by niik** [comment]
We should discuss whether to hold this until after 1.1. It's a large change and I'd like to have it sit on beta for at least a little while before we ship it.

**Comment by shiftkey** [comment]
@niik yep, that was my plan. I'll move this into 1.2 to make it extra clear.

**Comment by niik** [comment]
Just saw [an issue about crashes on Windows 10 Fall Creator Update](https://github.com/electron/electron/issues/11871) in the electron repo. We should make sure we've tested it out on that particular Windows version if we haven't already.

**Comment by iAmWillShepherd** [comment]
I tested this on the latest version of Windows (full 2 hours of updates and restarts) and it seems to work just fine.

**Comment by shiftkey** [comment]
I've bumped this to match the package upgrades in #4198, and we should do another pass of testing as part of reviewing...

File: app/package.json [hunk 3]
```diff
@@ -58,7 +58,8 @@
   "devDependencies": {
     "devtron": "^1.4.0",
     "electron-debug": "^1.1.0",
-    "electron-devtools-installer": "^2.2.1",
+    "electron-devtools-installer": "^2.2.3",
+    "react-addons-perf": "15.4.2",
```
Comment (shiftkey) [inline app/package.json]: This dependency needs to be removed :fire:

**Comment by shiftkey** [comment]
> We should make sure we've tested it out on that particular Windows version if we haven't already.

I've tested Electron 1.8.3 against that specific build and it's working fine.

My gut feeling about that upstream issue is that it's related to using Electron within a test runner - we had an issue with `electron-mocha` ages ago https://github.com/desktop/desktop/pull/1430#issuecomment-302373025 which required a fix to the test runner itself https://github.com/jprichardson/electron-mocha/pull/105

**Comment by niik** [comment]
> I've tested Electron 1.8.3 against that specific build and it's working fine.

👍

**Pull request merged**